src/transformers/generation/logits_process.py

```diff
@@ -133,14 +133,52 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 class MinNewTokensLengthLogitsProcessor(LogitsProcessor):
     r"""
     [`LogitsProcessor`] enforcing a min-length of new tokens by setting EOS (End-Of-Sequence) token probability to 0.
+    Note that for decoder-only models, such as Llama2, `min_length` will compute the length of `prompt + newly
+    generated tokens` whereas for other models it will behave as `min_new_tokens`,that is, taking only into account the
+    newly generated ones.
 
     Args:
         prompt_length_to_skip (`int`):
-            The input tokens length.
+            The input tokens length. Not a valid argument when used with `generate` as it will automatically assign the
+            input length.
         min_new_tokens (`int`):
             The minimum *new* tokens length below which the score of `eos_token_id` is set to `-float("Inf")`.
         eos_token_id (`Union[int, List[int]]`):
             The id of the *end-of-sequence* token. Optionally, use a list to set multiple *end-of-sequence* tokens.
+
+    Examples:
+
+    ```python
+    >>> from transformers import AutoTokenizer, AutoModelForCausalLM
+
+    >>> tokenizer = AutoTokenizer.from_pretrained("distilgpt2")
+    >>> model = AutoModelForCausalLM.from_pretrained("distilgpt2")
+    >>> model.config.pad_token_id = model.config.eos_token_id
+    >>> model.generation_config.pad_token_id = model.config.eos_token_id
+    >>> input_context = "Hugging Face Company is"
```

Comment:
Could we use a different input string e.g. "Hugging Face is"? 

The examples are a bit confusing because of `"Company"` being in the input and then `"company"` being set as the eos token

Reply:
Ahh, yes that makes sense, good catch :sparkles:

Reply:
Hi @amyeroberts I've been trying several prompts to no luck. I'm finding kind of hard getting two instances of the same token when one buries the first occurrence like in the 3rd example (blame my lack of experience :sweat_smile:). My guess is that `Company` is promoting `company` somehow. The workaround I found is to use a couple of `eos_token_ids` which has some educational value on its own. This is what happens for the prompt  `Hugging Face Inc. is`:

- `eos_token_ids=1664, min_new_tokens=1`: Hugging Face Inc. is a company 
- `eos_token_ids=[1664, 9856], min_new_tokens=2`: Hugging Face Inc. is a non-profit organization that provides educational
- `eos_token_ids=[1664, 9856], min_new_tokens=10`: Hugging Face Inc. is a non-profit organization that provides free, open source software to the

I'm happy to go with above examples, but I kind of like the existing ones as they only involve the change in one parameter, ie, the `min_new_tokens` so it's more like a RCT. In that case, what we can do is to clarfy this comment:
 ```
If `eos_token_id` is set to ` company` it will take into account how many `min_new_tokens` have 
been generated before stopping.
```
To something like:
```
If `eos_token_id` is set to ` company` it will take into account how many `min_new_tokens` have 
been generated before stopping. Note that ` Company` (5834) and ` company` (1664) are not 
actually the same token, and even if they were ` Company` would be ignored by `min_new_tokens`
as it excludes the prompt.
```
Let me know what you think 🙂

Reply:
Updated comment clarifying what's happening sounds good to me!

Reply:
Nice, thanks for your patience :pray: